bluemira/base/components.py

```diff
@@ -107,21 +108,74 @@ def get_component(
 
         Notes
         -----
-            This function is just a wrapper of the anytree.search.findall_by_attr
+            This function is just a wrapper of the anytree.search.findall
             function.
         """
-        if full_tree:
-            found_components = anytree.search.findall_by_attr(self.root, name)
-        else:
-            found_components = anytree.search.findall_by_attr(self, name)
+        return self._get_thing(
+            lambda n: anytree.search._filter_by_name(n, "name", name), first, full_tree
```

Comment:
I know this is using a private function but it seems a bit ott to rewrite what is a 3 line function that is used as the filter in `findall_by_attr` anyway

Reply:
I guess I'm OK with this, `get_component` is pretty well tested, so we should get an early warning if this API changes or disappears.